### ISSUE TITLE
[BUG] Problem syncing with FeedMe

### ISSUE THREAD

**Issue by tonitonae** [FreshRSS/FreshRSS#4618]
Hi.

Yesterday I upgraded to 1.20.0 (ARM, on a RPi, on Docker). I come from 1.1X (do not remember, sorry, something like 15 months old... yes, I know).

I use FeedMe to read my feeds in my mobile phone. Until yesterday, syncing worked fine. However, after upgrading, most of the times, after I hit the sync button, nothing happens. However, if I log out and in again, feeds are synced again, but this does not last. With a high probability, when I hit the sync button again it will not sync.

**To Reproduce**
1. Login to FreshRSS (1.20.0) from FeedMe.
2. Click on the sync button.
3. Nothing happens (most of the times).

**Expected behavior**
Updates to the feeds should sync to FeedMe.

**Environment information (please complete the following information):**
- Device: Raspberry Pi
- OS: DietPi (Debian based)
- Browser: Android client (FeedMe)
- FreshRSS version: 1.20.0
- Database version: SQLite
- PHP version: do not know, I am using official Docker image
- Installation type: Docker

**Additional context**
What follows is what I can see in Docker logs. First 4 lines are output when sync does not work. The next 23 lines are what is output when sync works. These logs were obtained one after the other: I hit once, nothing happened, I hit a second time, it synced. However, normally this is not the case: I can hit several times and it does not sync.

I post this here because, maybe, something changed in the API and makes this happen. FeedMe is updated to the last version, which was released several months ago.

```
192.168.1.65 - - [16/Sep/2022:15:52:51 +0200] "GET /api/greader.php/reader/api/0/tag/list?output=json HTTP/1.1" 200 137 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:52:51 +0200] "GET /api/greader.php/reader/api/0/subscription/list?output=json HTTP/1.1" 200 1705 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:11 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=500&ot=0&s=user%2F-%2Fstate%2Fcom.google%2Fstarred HTTP/1.1" 200 16 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:52:51 +0200] "GET /api/greader.php/reader/api/0/unread-count?output=json HTTP/1.1" 200 481 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"

192.168.1.65 - - [16/Sep/2022:15:53:48 +0200] "GET /api/greader.php/reader/api/0/tag/list?output=json HTTP/1.1" 200 137 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:48 +0200] "GET /api/greader.php/reader/api/0/subscription/list?output=json HTTP/1.1" 200 1705 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:48 +0200] "GET /api/greader.php/reader/api/0/unread-count?output=json HTTP/1.1" 200 481 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:49 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=95&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663294560&s=feed%2F14 HTTP/1.1" 200 490 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:49 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=94&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663321320&s=feed%2F35 HTTP/1.1" 200 1916 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:49 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=91&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663325580&s=feed%2F40 HTTP/1.1" 200 96 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=91&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663319820&s=feed%2F29 HTTP/1.1" 200 10516 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=86&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663319040&s=feed%2F30 HTTP/1.1" 200 2172 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=79&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663311120&s=feed%2F39 HTTP/1.1" 200 13800 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=71&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663323780&s=feed%2F34 HTTP/1.1" 200 2032 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=65&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663318800&s=feed%2F36 HTTP/1.1" 200 35961 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/contents?output=json&n=54&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=1663324200&s=feed%2F26 HTTP/1.1" 200 2851 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F14 HTTP/1.1" 200 41 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F35 HTTP/1.1" 200 98 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F40 HTTP/1.1" 200 98 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F29 HTTP/1.1" 200 106 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F30 HTTP/1.1" 200 92 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F39 HTTP/1.1" 200 87 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F34 HTTP/1.1" 200 151 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F36 HTTP/1.1" 200 156 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=20&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&s=feed%2F26 HTTP/1.1" 200 206 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=15&xt=user%2F-%2Fstate%2Fcom.google%2Fread&ot=0&c=1663295218134215&s=feed%2F26 HTTP/1.1" 200 180 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
192.168.1.65 - - [16/Sep/2022:15:53:50 +0200] "GET /api/greader.php/reader/api/0/stream/items/ids?output=json&n=500&ot=0&s=user%2F-%2Fstate%2Fcom.google%2Fstarred HTTP/1.1" 200 16 "-" "FeedMe/3.16 (com.seazon.feedme; build:206; Android SDK 30)"
```

Thanks for your help.

**Comment by Alkarex** [comment]
Hello,
I have tried with our FreshRSS 1.20.0 with Docker on Rasperry Pi, using Feedme v3.20.14 and cannot spot any problem (besides the fact that Feedme's synchronisation strategy is still not optimal - cf. my recommendation https://github.com/FreshRSS/FreshRSS/issues/2566#issuecomment-541317776 ).

Make sure that in the Feedme settings, your synchronisation mode is set to *Everything*. When testing, try to mark an article in FreshRSS as unread from the Web interface.

**Comment by tonitonae** [comment]
Hi @Alkarex.

Thanks for your prompt reply.

I have been doing some testing and it seems that, even having the "Everything" option selected in FeedMe, feeds do not updated unless there are some articles marked as read from the Web interface.

Say that I have 50 unread posts in my feeds. If FreshRSS fetches another 20 (up to a total of 70), if I hit the update button in FeedMe, nothing happens in the app, even if can see that there are 70 posts in the web interface.

If I mark a couple of messages as read in the web interface and then sync from the FeedMe client, then new messages are synced to the app.

I can repeat the process, it seems consistent: I have now 67 messages (I marked 3 as read in the web interface), 10 more arrive (77 in total) and I hit the update button in FeedMe. Nothing happens. I open the web interface, read 5 more messages, 72 remaining. Open FeedMe again and 72 posts get synced.

I have repeated all this 2-3 times and it seems to be consistent. I don not know, however, why this is happening and what should I touch either in FreshRSS or FeedMe to get back a working sync.

I will continue testing, but any suggestions are welcome.

Thanks!

**Comment by Alkarex** [comment]
I cannot reproduce any problem. For debugging, can you try with another app such as https://github.com/readrops/Readrops ?

**Comment by tonitonae** [comment]
Hi @Alkarex .

I have been doing some extra testing these days.

FeedMe seems to be stuck for some reason. No matter how many times I try to sync, it does not update the feeds. This is weird as it used to be working before the update to FreshRSS, but the only way to make it sync is to logout and ind (and this only lasts some hours).

I have tried Readrops, and it works, but the app has not been updated for the last 12 months and I miss some functionality from FeedMe.

I have now installed Focus Reader and it also seems to be working (and syncs are blazing fast compared to FeedMe). It has all the functionality that I used from FeedMe so I am going to switch to it for the moment.

I think that we can close the issue, although I do not really understand what has happened and why FeedMe has just stopped working all of a sudden (for me, at least).

Thanks for your help and suggestions.
